pkg/util/certificates/csr.go

```diff
@@ -38,3 +45,86 @@ func ParseCertificateRequestObject(obj *certificates.CertificateSigningRequest)
 	}
 	return csr, nil
 }
+
+// NewCertificateRequest generates a PEM-encoded CSR using the supplied private
+// key data, subject, and SANs. If the private key data is empty, it generates a
+// new ECDSA P256 key to use and returns it together with the CSR data.
+func NewCertificateRequest(keyData []byte, subject *pkix.Name, dnsSANs []string, ipSANs []net.IP) (csr []byte, key []byte, err error) {
```

Comment:
let's split this into three methods, something like https://github.com/liggitt/kubernetes/blob/tls-bootstrap/pkg/util/certificates/csr.go#L49-L139

```
// GeneratePrivateKey returns PEM data containing a generated ECDSA private key
func GeneratePrivateKey() ([]byte, error)

// ParsePrivateKey returns a private key parsed from a PEM block in the supplied data.
// Recognizes PEM blocks for "EC PRIVATE KEY" and "RSA PRIVATE KEY"
func ParsePrivateKey(keyData []byte) (interface{}, error)

// NewCertificateRequest generates a PEM-encoded CSR using the supplied private key, subject, and SANs.
// privateKey must be a *ecdsa.PrivateKey or *rsa.PrivateKey.
func NewCertificateRequest(privateKey interface{}, subject *pkix.Name, dnsSANs []string, ipSANs []net.IP) (csr []byte, err error)
```

that lets each function be small and individually useful, and lets us avoid the passthrough keydata param -> return value